### ISSUE TITLE
Run Retinanet on my own object detection dataset?

### ISSUE THREAD

**Issue by allansdefreitas** [keras-team/keras-io#946]
Hello,

I'm running this Keras implementation of [retinanet](https://github.com/keras-team/keras-io/blob/master/examples/vision/retinanet.py) and it works perfectly on COCO dataset obtained from tfds.load().

I noticed that the downloaded TFrecords (from tfds.load() ) contain COCO annotations in a different way from traditional COCO annotations and that is the cause of all problems. Please, see: [annotation comparison](https://drive.google.com/file/u/1/d/1sFIQ-LeB88zUDxNPEfcJjkNeYopu9ZLb/view?usp=sharing)

**The problem is: How to run this model on my own dataset? i.e.
How to put my dataset in this specific COCO annotation format?**

Thanks in advance,
Allan Freitas

**Comment by ojay37** [comment]
@allansdefreitas 

You will need to create your own generator that loads your specific dataset into the format that is expected. 

Your generator will need to return 3 separate tensors per image. These will represent image, bounding boxes, and classes , respectively. 

If memory serves me right, the image should be shape [X, Y, 3] where the first two indices are x, y and divisible by 32.
Bounding boxes is an array of bounding boxes, which have dimensions of X, Y, Width, Height. -- Representing the anchor box co-ordinates.

And, the classes is an array of class Ids corresponding to the bounding boxes.

**Comment by yassiney** [comment]
> How to put my dataset in this specific COCO annotation format?

How to put my dataset in this specific COCO annotation format?
They are normalized annotations:  `**Y / image height**` and `**X / image width**`

**Comment by pcoet** [comment]
Hi folks, thanks for the discussion. The issue backlog is not the best place for support requests. Can you repost this to one of the forums below? That way your discussion will have more visibility.

* [TensorFlow forum](https://discuss.tensorflow.org/)
* [Keras mailing list](https://groups.google.com/forum/#!forum/keras-users)

Thanks!
